fabric-biomes-v1/src/main/java/net/fabricmc/fabric/mixin/biomes/VanillaLayeredBiomeSourceMixin.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.biomes;
+
+import java.util.Set;
+import java.util.function.Function;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.fabricmc.fabric.impl.biomes.BiomeLists;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.source.VanillaLayeredBiomeSource;
+import net.minecraft.world.gen.feature.StructureFeature;
+
+@Mixin(VanillaLayeredBiomeSource.class)
+public class VanillaLayeredBiomeSourceMixin
+{
+	@Inject(at = @At("HEAD"), method = "hasStructureFeature", cancellable = true)
+	private void hasStructureFeature(StructureFeature<?> structureFeature_1, CallbackInfoReturnable<Boolean> info)
+	{
+		Function<StructureFeature<?>, Boolean> b = (structureFeature_1x) -> {
+			
+			Set<Biome> var2 = BiomeLists.CUSTOM_BIOMES;
+			
+			for(Biome biome_1 : var2)
```

Comment:
Those variables could have proper names too, copy-pasted decompiled names in the official API doesn't look too good.

Reply:
I'll remap them

Reply:
Tbh some of them I made randomly because I had no idea what to name them

Like int_66 and int_42 in the hills layer mixin

Reply:
If you can think of better names for any of them let me know or something